functions.go

```diff
@@ -115,7 +115,7 @@ func init() {
 	funcs.register("fact", &function{
 		arity: 1,
 		fn: func(args []float64) float64 {
-			return Factorial(args[0])
+			return float64(Factorial(int64(args[0] + 0.5)))
```

Comment:
Any reason you're adding 0.5 here instead of just flooring or ceiling? (Or just casting it to an `int64`, which will floor it automatically)